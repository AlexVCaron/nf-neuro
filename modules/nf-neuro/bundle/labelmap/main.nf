process BUNDLE_LABELMAP {
    tag "$meta.id"
    label 'process_single'

    container "scilus/scilpy:2.2.0_cpu"

    input:
        tuple val(meta), path(bundles), path(centroids)

    output:
    tuple val(meta), path("*_labels.nii.gz")    , emit: labels
    tuple val(meta), path("*_labels.trk")       , emit: labels_trk
    tuple val(meta), path("*_distances.nii.gz") , emit: distances
    tuple val(meta), path("*_distances.trk")    , emit: distances_trk
    path "versions.yml"                         , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def nb_points = task.ext.nb_points ? "--nb_pts ${task.ext.nb_points} ": ""
    def colormap = task.ext.colormap ? "--colormap ${task.ext.colormap} ": ""

    """
    bundles=(${bundles.join(" ")})
    centroids=(${centroids.join(" ")})

    for index in \${!bundles[@]};
        do ext=\${bundles[index]#*.}
        bname=\$(basename \${bundles[index]} .\${ext})

        scil_bundle_label_map \${bundles[index]} \${centroids[index]} \
            tmp_out $nb_points $colormap -f

        mv tmp_out/labels_map.nii.gz ${prefix}__\${bname}_labels.nii.gz
        mv tmp_out/distance_map.nii.gz ${prefix}__\${bname}_distances.nii.gz
        mv tmp_out/labels.trk ${prefix}__\${bname}_labels.trk
        mv tmp_out/distance.trk ${prefix}__\${bname}_distances.trk
    done

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(uv pip -q -n list | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    scil_bundle_label_map -h

    bundles=(${bundles.join(" ")})
    for index in \${!bundles[@]};
        do ext=\${bundles[index]#*.}
        bname=\$(basename \${bundles[index]} .\${ext})

        touch ${prefix}__\${bname}_labels.nii.gz
        touch ${prefix}__\${bname}_labels.trk
        touch ${prefix}__\${bname}_distances.nii.gz
        touch ${prefix}__\${bname}_distances.trk
    done

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(uv pip -q -n list | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """
}
