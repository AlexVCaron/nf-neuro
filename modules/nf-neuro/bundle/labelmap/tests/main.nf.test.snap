{
    "bundle - labelmap - stub-run": {
        "content": [
            [
                "versions.yml:md5,c126a7422f280ed6bd664cc0f6636931"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-06-03T17:00:40.763715496"
    },
    "bundle - labelmap": {
        "content": [
            [
                "versions.yml:md5,c126a7422f280ed6bd664cc0f6636931"
            ],
            "test__IFGWM_labels.nii.gz",
            "test__IFGWM_labels.trk",
            "test__IFGWM_distances.nii.gz",
            "test__IFGWM_distances.trk"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-05-22T16:31:38.189887"
    }
}
